cmd/gh/main.go

```diff
@@ -58,14 +59,29 @@ func printError(out io.Writer, err error, cmd *cobra.Command, debug bool) {
 		return
 	}
 
+	var flagError *command.FlagError
+	if errors.As(err, &flagError) {
+		if flagError.Err == pflag.ErrHelp {
```

Comment:
Thank you for letting us know about the error!

The additions in this PR feel like they add too much special `pflag.ErrHelp` handling considering that Cobra was already supposed to handle that flag internally. It looks like we broke that handling by wrapping this error in the first place. I've opened an alternate PR to explore a simpler fix: https://github.com/cli/cli/pull/574